content/stm32/flash-cube-programmer.md

```diff
@@ -1,25 +1,47 @@
+This is a "rough" guide on how to manually flash the firmware using STM32CubeProgrammer utility.
 
-You will need to use this utility to replace the boot loader on the STM32 chip.  Once replaced, you can do all further firmware maintentance using the nanoFramework Firmware Flash tool (nanoff.exe).
+NOTE: This utility replaces the deprecated ST-Link utility and is required as a prerquisite for boards with outdated firmware. Outdated boards will need to use this utility to update their current boot loader firmware. Once updated, all further firmware maintentance can be carried out using the nanoFramework Firmware Flash tool [nanoff](https://github.com/nanoframework/nanoFirmwareFlasher).
 
-## Install the tool
+## Install the STM32 Cube Programmer utility
 
-1. Download the appropriate [STM32 Cube Programmer](https://www.st.com/en/development-tools/stm32cubeprog.html) from ST web site and install it in your development machine.
+1. Download the appropriate [STM32 Cube Programmer](https://www.st.com/en/development-tools/stm32cubeprog.html) from ST website and install it on your development machine.
 
-## Flash a JTAG connected board
+## Get the .NET nanoFramework device Firmware
 
-There are two images to be flashed in the target, one for nanoBooter and another one for nanoCLR.
+There are two firmware binaries to be flashed to the device, one for nanoBooter and another one for nanoCLR.
 
-1. Download a ZIP file with the firmware for the board from our web site [here](https://github.com/nanoframework/nf-interpreter#firmware-for-reference-boards) by clicking on the appropriate badge. This will take you to our Cloudsmith repository that holds the packages with pre-build images for several target boards. After downloading it, unzip the package contents.
+1. Download the binaries for our reference boards [here](https://github.com/nanoframework/nf-interpreter#firmware-for-reference-boards) by clicking on the appropriate badge. This will take you to our Cloudsmith repository that holds the packages with pre-build images. After downloading it, ensure to unzip the package contents to a folder of your choosing.
 
-1. Connect the target board to your PC using an USB cable. Note that on most ST development boards there are two micro USB connectors. To follow this guide you'll want to use the one that's providing the JTAG connection through the ST-Link debugger. In doubt, check the PCB for the correct one or the board schematic.
+## Setup Connection
 
 1. Launch the STM32 Cube Programmer that you've just installed and connect to the ST board.
 
+### Connect to a ST-Link connected board
+
+Note: ensure you Update the STLink Firmware before continuing using the `Firmware Upgrade` button and following the instructions.
+
+1. Select "ST-Link" in the interface options.
+    >![ST-LINK](../../images/stm32/stm32-cube-programmer-select-stlink.jpg)
+
+### Connect to a JTAG connected board
```

Comment:
Guess you're referring to boards such as ORGPAL_3 that require an external ST-Link pod.
If that's the case, then the paragraph below should be on the one before this and here the one above with the recommendation to update the STLink fw.

Reply:
You've expanded this explanation to 3 cases: JTAG, STLing and DFU. 
I was guessing that you're referring to JTAG connected boards to ones similar to ORGPAL that require an external JTAG pod, as opposed to the STM32 Nucleo and Discovery ones that have it onboard.